src/native/libs/System.Net.Security.Native/pal_gssapi.c

```diff
@@ -183,10 +203,19 @@ static uint32_t AcquireCredSpNego(uint32_t* minorStatus,
 
     // call gss_set_cred_option with GSS_KRB5_CRED_NO_CI_FLAGS_X to support Kerberos Sign Only option from *nix client against a windows server
 #if HAVE_GSS_KRB5_CRED_NO_CI_FLAGS_X
-    if (majorStatus == GSS_S_COMPLETE)
+    if (majorStatus == GSS_S_COMPLETE && GSS_KRB5_CRED_NO_CI_FLAGS_X_AVAILABLE)
     {
         GssBuffer emptyBuffer = GSS_C_EMPTY_BUFFER;
-        majorStatus = gss_set_cred_option(minorStatus, outputCredHandle, GSS_KRB5_CRED_NO_CI_FLAGS_X, &emptyBuffer);
+        uint32_t tempMinorStatus;
+        majorStatus = gss_set_cred_option(&tempMinorStatus, outputCredHandle, GSS_KRB5_CRED_NO_CI_FLAGS_X, &emptyBuffer);
+        if (majorStatus == GSS_S_UNAVAILABLE)
+        {
+            majorStatus = GSS_S_COMPLETE;
+        }
+        else
+        {
+            *minorStatus = tempMinorStatus;
+        }
```

Comment:
This can be simplified, since minorStatus value is unused.
```suggestion
        majorStatus = gss_set_cred_option(&minorStatus, outputCredHandle, GSS_KRB5_CRED_NO_CI_FLAGS_X, &emptyBuffer);
        if (majorStatus == GSS_S_UNAVAILABLE)
        {
            majorStatus = GSS_S_COMPLETE;
        }
```

(same as below)

Reply:
It is not unused, it is passed to the managed code (where it is indeed unused but that may change).

Reply:
In this particular case, it's problematic to change the native prototypes because dotnet/sqlclient has a dependency on it. We are trying to improve the situation going forward by introducing the `NegotiateAuthentication` public API but that work didn't land yet.

Reply:
As minimum, I would suggest to put the minor error code to tracing. e.g. make use of it for debugging.

Reply:
I double-checked and it's passed in exception already:
https://github.com/dotnet/runtime/blob/6de7147b9266d7730b0d73ba67632b0c198cb11e/src/libraries/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs#L126

Reply:
Your proposed change always overrides the `minor_status` from `gss_acquire_cred` if this code path is taken. The idea was to preserve full `gss_acquire_cred` status if `gss_set_cred_option` returns `GSS_S_UNAVAILABLE` or `GSS_S_COMPLETE`.

Apparently somewhere along the way I incorrectly dropped the `GSS_S_COMPLETE` from the first `if`. I'll fix that.

Reply:
Note that fixing the `if` makes no difference since the managed code ignores `minorStatus` when the major one is `GSS_S_COMPLETE` but I want to make the intention clear. That is, preserve the full `gss_acquire_cred` status if `gss_set_cred_option` succeeds or is unavailable. In all other cases return the full status from `gss_set_cred_option`.